src/Tools/dotnet-dump/ExtensionCommands/ExtensionCommandBase.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Repl;
+
+namespace Microsoft.Diagnostic.Tools.Dump.ExtensionCommands
+{
+    public abstract class ExtensionCommandBase : CommandBase
+    {
+        /// <summary>
+        /// Helper bound to the current ClrRuntime that provides
+        /// high level services on top of ClrMD.
+        /// </summary>
+        public ClrMDHelper Helper { get; set; }
+
+        [HelpInvoke]
+        public void InvokeHelp()
```

Comment:
If you always define a HelpInvoke/InvokeHelp() function, your commands will always have to supply the help text because the command processor doesn't print the "default" help text built by System.CommandLine in that case.

Reply:
The default help looks something like this:

```
> help clrmodules
clrmodules:
  Lists the managed modules in the process.

Usage:
  > clrmodules [options]

Options:
  -v, --verbose    Displays detailed information about the modules.
```

Reply:
> If you always define a HelpInvoke/InvokeHelp() function, your commands will always have to supply the help text because the command processor doesn't print the "default" help text built by System.CommandLine in that case.

I'm making GetDetailedHelp abstract to force all command to implement it and avoid this kind of error if a command forget to provide help


> The default help looks something like this:

I have used the SOS way of showing help 
ex for dumpobj in dotnet-dump (BTW, contains the WinDBG ! prefix due to the fact that the help command is directly proxied to sos itself):
```
> help dumpobj
-------------------------------------------------------------------------------
!DumpObj [-nofields] <object address>

This command allows you to examine the fields of an object, as well as learn
important properties of the object such as the EEClass, the MethodTable, and
the size.

You might find an object pointer by running !DumpStackObjects and choosing
from the resultant list. Here is a simple object:

    0:000> !DumpObj a79d40
    Name: Customer
    MethodTable: 009038ec
    EEClass: 03ee1b84
    Size: 20(0x14) bytes
     (C:\pub\unittest.exe)
    Fields:
          MT    Field   Offset                 Type  VT     Attr    Value Name
    009038ec  4000008        4             Customer   0 instance 00a79ce4 name
    009038ec  4000009        8                 Bank   0 instance 00a79d2c bank
```

Reply:
I was just giving you an example of the default help not criticizing your formatting.  And making the function abstract is a good solution.